python/lib/core/dmod/core/context/monitor.py

```diff
@@ -0,0 +1,388 @@
+        return self.__logger
+
+    @logger.setter
+    def logger(self, logger: LoggerProtocol) -> None:
+        self.__logger = logger
+
+    @property
+    def size(self) -> int:
+        return self.__size
+
+    def __len__(self):
+        return self.size
+
+    @property
+    def __should_be_monitoring(self) -> bool:
+        """
+        Whether the monitor should still be polling
+
+        This should monitor if it's either accepting items to monitor or it has items left to iterate through
+        """
+        with self.__lock:
+            return self.accepting_futures or self._queue.qsize() != 0
+
+    def _monitor(self) -> bool:
+        """
+        Loop through the job results in the queue and delete them when processing has completed
+
+        Returns:
+            True if the function ended with no issues
+        """
+        monitoring_succeeded: bool = True
+
+        while self.__should_be_monitoring:
+            try:
+                # Block to check if the loop should be exitted based on a current kill state
+                with self.__lock:
+                    if self.monitor_should_be_killed:
+                        monitoring_succeeded = False
+                        break
+
+                    future_result: typing.Union[JobResultProtocol, object] = self._queue.get(
+                        timeout=self._timeout
+                    )
+                    self.__size -= 1
+
+                if future_result in (self._PING_SIGNAL, self._KILL_SIGNAL, self._STOP_SIGNAL):
+                    if future_result == self._KILL_SIGNAL:
+                        monitoring_succeeded = False
+                        self.__killed = True
+                        break
+
+                    if future_result == self._STOP_SIGNAL:
+                        self.__stopping = True
+                    continue
+
+                # This is just junk if it isn't a job result, so acknowledge it and move to the next item
+                if not isinstance(future_result, JobResultProtocol):
+                    self.logger.error(
+                        f"Found an invalid value in a {self.class_name}:"
+                        f"{future_result} must be either a future or {self.class_name}._PING_SIGNAL, "
+                        f"{self.class_name}._STOP_SIGNAL, or {self.class_name}._KILL_SIGNAL, "
+                        f"but received a {type(future_result)}"
+                    )
+                    continue
+
+                # If the process is done, fetch the result, call the callback if it didn't error,
+                #   record the error if it did, and clean up any associated scope information
+                if future_result.done():
+                    scope = self.find_scope(future_result)
+
+                    try:
+                        value = future_result.result()
+                        if self._callback:
+                            try:
+                                self._callback(value)
```

Comment:
As above:

Failing here would kill the runner. Evaluation B may have a specific configuration that throws an error, but evaluations A, C, D, E, etc may not engage in that faulty functionality. This leaves an opportunity where a single instance of a malformed json string may knock the entire application offline.

Additionally, since this is a tracking thread, stopping the thread here could yield seemingly unrelated errors elsewhere due to reference counters among interconnected processes that are hard to track and diagnose.

We don't want any of that, so we record the error and move on. Imagine an error here being along the lines of an HTTP 500 error.